incubating/service-now/step.yaml

```diff
@@ -0,0 +1,139 @@
+kind: step-type
+version: '1.0'
+metadata:
+  name: service-now
+  version: 1.0.0
+  isPublic: true
+  description: Integration with ServiceNow Change Management
```

Comment:
This is a fairly complex step that can wait for a callback. Considering adding a sentence here to describe what all it does.